src/libraries/Common/src/System/Net/SocketProtocolSupportPal.cs

```diff
@@ -12,7 +13,10 @@ internal static partial class SocketProtocolSupportPal
 
         public static bool OSSupportsIPv6 { get; } = IsSupported(AddressFamily.InterNetworkV6) && !IsIPv6Disabled();
         public static bool OSSupportsIPv4 { get; } = IsSupported(AddressFamily.InterNetwork);
-        public static bool OSSupportsUnixDomainSockets { get; } = IsSupported(AddressFamily.Unix);
+        // iOS/tvOS ostensibly has AF_UNIX, but throws EPERM on iOS/tvOS 10.0+ on bind
+        // Mac Catalyst returns true on IsIOS, so we need to explicitly check for the case
+        // "is iOS but not Catalyst" as unsupported, but Catalyst itself is supported.
+        public static bool OSSupportsUnixDomainSockets { get; } = IsSupported(AddressFamily.Unix) && !OperatingSystem.IsTvOS() && (!OperatingSystem.IsIOS() || OperatingSystem.IsMacCatalyst());
```

Comment:
```suggestion
        public static bool OSSupportsUnixDomainSockets { get; } = !OperatingSystem.IsTvOS() && (!OperatingSystem.IsIOS() || OperatingSystem.IsMacCatalyst()) && IsSupported(AddressFamily.Unix);
```

The operating system checks are essentislly free and we can entirely skip the call to `IsSupported` on unsupported platforms.